Doc/library/typing.rst

```diff
@@ -1484,6 +1481,18 @@ These are not used in annotations. They are building blocks for declaring types.
       The keyword-argument syntax is deprecated in 3.11 and will be removed
       in 3.13. It may also be unsupported by static type checkers.
 
+   The functional syntax should also be used when any of the keys are not valid
+   :ref:`identifiers`, for example because they are keywords or contain hyphens.
```

Comment:
```suggestion
   :ref:`identifiers`.
```

There can be other reason:
```py
from typing import TypedDict

A = TypedDict("A", {"a b c": int})
a = A()
a["a b c"] = 1
```

Reply:
The "for example" wording does not preclude the possibility that there are other reasons; it just gives examples of _some_ of the reasons

Reply:
Ya, but `not a valid identifier` is enough and no need to give examples.

Reply:
Hmm, I disagree — I don't think it's self-evident what the definition of an "identifier" is, and the link takes you (I think — haven't checked) to an extremely jargony page in the docs that's pretty scary tbh!

Reply:
Okay I agree now